src/components/modal/__internal__/modal-manager.js

```diff
@@ -1,26 +1,54 @@
 class ModalManagerInstance {
   #modalList = [];
 
-  addModal = (modal) => {
-    this.#modalList.push(modal);
+  #getTopModal() {
+    if (!this.#modalList.length) {
+      return {};
+    }
+    return this.#modalList[this.#modalList.length - 1];
+  }
+
+  addModal = (modal, setTriggerRefocusFlag) => {
+    const {
+      modal: topModal,
+      setTriggerRefocusFlag: setTrapFlag,
+    } = this.#getTopModal();
+
+    if (topModal && setTrapFlag) {
+      setTrapFlag(false);
+    }
+
+    this.#modalList.push({ modal, setTriggerRefocusFlag });
   };
 
   isTopmost(modal) {
-    if (!modal || !this.#modalList.length) {
+    if (!modal || !this.#getTopModal().modal) {
       return false;
     }
+    const { modal: topModal } = this.#getTopModal();
 
-    return this.#modalList.indexOf(modal) === this.#modalList.length - 1;
+    return modal === topModal;
   }
 
   removeModal(modal) {
-    const modalIndex = this.#modalList.indexOf(modal);
+    const modalIndex = this.#modalList.findIndex(({ modal: m }) => m === modal);
 
     if (modalIndex === -1) {
       return;
     }
 
     this.#modalList.splice(modalIndex, 1);
+
+    if (!this.#modalList.length) {
+      return;
+    }
+
+    const { modal: topModal, setTriggerRefocusFlag } = this.#getTopModal();
+
+    /* istanbul ignore else */
+    if (topModal && setTriggerRefocusFlag) {
```

Comment:
Saying as we've already returned out the function earlier if `this.#modalList.length` - it shouldn't be possible to get to this line and `topModal` to not be set so we can remove that from the if statement.

A test should be added for when `setTriggerRefocusFlag` is falsey rather than ignoring the coverage here

Reply:
Yeah good point, I'll update the if statement and write and else test 👍